src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -57,6 +61,12 @@ typedef struct {
    *  currently assigned. If the actor process reports back success for the
    *  assigned task execution, task_counter should be set to this value. */
   int64_t assigned_task_counter;
+  /** Whether the actor process has loaded yet. The actor counts as loaded once
+   *  it has either executed its first task or successfully resumed from a
+   *  checkpoint. Before the actor has loaded, we may dispatch the first task
+   *  or any checkpoint tasks. After it has loaded, we may only dispatch tasks
+   *  in order. */
+  bool loaded;
```

Comment:
Ok, I think I was a bit confused. I thought the `i`th checkpoint task loaded the `i`th checkpoint, but actually it loads the most recent checkpoint, is that right?

Reply:
Hmm, I think it will be okay since only certain tasks will get resubmitted during recovery. Earlier checkpoint tasks will fail since their checkpoint isn't available. It is possible that the constructor could run first, but that will only happen if someone actually needs the results of a task that happened before the checkpoint. Otherwise, the task won't get resubmitted.

For the latter case, it could be good to check for a checkpoint in the constructor (similar to what you had before). I'll add a TODO.

Nope, it does load the `i`th checkpoint, but only if it exists. Else, it'll set the `actor_checkpoint_failed` field in the response to the local scheduler.